### PR TITLE
Add option to disable players respawning at beds

### PR DESCRIPTION
**Pull request by mdcfe**
### Information

This PR addresses a request made on Discord but doesn't close an existing issue.

### Details

**Proposed feature**  
Add an option to toggle whether beds are considered valid respawn points in EssentialsX Spawn.

**Environments tested:**  
- [ ] [Latest](https://papermc.io/downloads) Paper Version (any OS, any Java 8+ version)
- [ ] CraftBukkit/Spigot/Paper 1.12.2 (any OS, any Java 8+ version)
- [ ] CraftBukkit 1.8.8 (any OS, any Java 8+ version)

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/Settings.java [hunk 1]
```diff
@@ -129,6 +129,7 @@
     private Set<Predicate<String>> nickBlacklist;
     private double maxProjectileSpeed;
     private boolean removeEffectsOnHeal;
+    private boolean bedSpawnEnabled;
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/Settings.java]:
```suggestion
```

File: Essentials/src/main/java/com/earth2me/essentials/Settings.java [hunk 2]
```diff
@@ -650,6 +651,7 @@ public void reloadConfig() {
         removeEffectsOnHeal = _isRemovingEffectsOnHeal();
         vanishingItemPolicy = _getVanishingItemsPolicy();
         bindingItemPolicy = _getBindingItemsPolicy();
+        bedSpawnEnabled = _isBedSpawnEnabled();
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/Settings.java]:
```suggestion
```

File: Essentials/src/main/java/com/earth2me/essentials/Settings.java [hunk 3]
```diff
@@ -1707,4 +1709,14 @@ public boolean isConfirmHomeOverwrite() {
...
+
+    private boolean _isBedSpawnEnabled() {
+        return config.getBoolean("respawn-at-home-bed", true);
+    }
+
+    @Override
+    public boolean isRespawnAtBed() {
+        return bedSpawnEnabled;
+    }
+
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/Settings.java]:
```suggestion
    @Override
    public boolean isRespawnAtBed() {
        return config.getBoolean("respawn-at-home-bed", true);
    }
```

**Review by JRoy — CHANGES_REQUESTED** [review]
Should only be caching config options ourselves if we mutate it in any way, which we don't here.

**Comment by mdcfe** [comment]
> Should only be caching config options ourselves if we mutate it in any way, which we don't here.

@JRoy Not really, this is the standard for Essentials' configs and minimises touching config objects when not reloading.

**Comment by JRoy** [comment]
@mdcfe The config object is already a cached representation of the config, this just creates a duplicated object in memory for no reason.
we already do this a lot in Settings

i should also make a follow up pr for removing all these when not needed

**Review by JRoy — APPROVED** [review]

**Pull request merged**